OpenLightingProject/ola
# GitHub Actions Lint: flake8, cpplint, enforce_licence, generic NOLINTs

Working toward #1815. This is still a draft. I am force-pushing to avoid massive commit spamming; I'll update the commit message to be very descriptive once I'm done.

I am using Debian stable as packaging on Debian seems to be very actively attended to by OLA contributors and not just Debian packagers. There also exist build instructions for Debian within the OLA repo.

So far:
- Build once, lint multiple times from that one build
- Don't require building first for lint tasks that don't need it
- Run flake8 the exact same way it would be run on a dev machine (from `Makefile`), but with annotations
- Move cpplint to latest `pip` version
- Add GH Actions annotations to `enforce_licence.py`
- Move the generic `NOLINT` check from the Travis-only script to a script that can be run anywhere
- Add GH Actions annotations to the generic `NOLINT` check

Still lots to do, but progress!
